vignettes/classification.Rmd

```diff
@@ -0,0 +1,260 @@
+| `-~`    | moderate negative effect | moderate decrease | confidence interval between reference and the lower threshold | 1 |
+| `-`    | negative effect | decrease | confidence interval below reference and contains the lower threshold | 1 |
+| `--`    | strong negative effect | strong decrease | confidence interval below the lower threshold | 0 |
+| `?+`    | potential positive effect  | potential increase  | confidence interval contains reference and the upper threshold | 2 |
+| `?-`    | potential negative effect  | potential decrease  | confidence interval contains reference and the lower threshold | 2 |
+| `?`    | unknown effect  | unknown  | confidence interval contains the lower and upper threshold | 3 |
+
+Table: Summary of the simple classification into ten categories.
+
+```{r detailed-effect3, fig.cap = "Graphical overview of the ten class classification split by the number of thresholds in the confidence interval.", echo = FALSE}
+library(effectclass, warn.conflicts = FALSE)
+ds <- data.frame(
+  thresholds = c( 3,  2,  1,  0,  2,  1,  0, 1, 0, 0),
+  lcl =        c(-5, -5, -5, -5, -2, -2, -2, 1, 1, 3),
+  ucl =        c( 5,  2, -1, -3,  5,  2, -1, 5, 2, 5)
+)
+ds$effect <- classification(ds$lcl, ds$ucl, threshold = 2.5)
+ds$coarse <- coarse_classification(ds$effect)
+ds$thresholds <- sprintf("interval contains\n%i thresholds", ds$thresholds)
+ggplot(ds, aes(ymin = lcl, ymax = ucl, x = effect)) +
+  geom_errorbar() +
+  geom_hline(yintercept = 0, linetype = 2) +
+  geom_hline(yintercept = c(-2.5, 2.5), linetype = 3) +
+  facet_wrap(~thresholds, scales = "free_x") +
+  scale_y_continuous(
+    breaks = c(-2.5, 0, 2.5), 
+    labels = c("lower\nthreshold", "reference", "upper\nthreshold")
+  ) +
+  theme(
+    axis.title = element_blank()
+  )
+```
+
+## Classification Using `effectclass`
+
+The `effectclass` package provides `classification()` for classifying confidence intervals.
+The function requires a vector of lower bounds of the confidence intervals (`ucl`) and a vector of upper bounds (`ucl`).
+It also requires the lower and upper thresholds (`threshold`) as a two-element vector.
+In case the thresholds are symmetric around the `reference`, you can provide a single value which is the absolute difference between the reference and both thresholds.
+The default `reference` is zero.
+`classification()` returns a ten level factor using the two digits symbols mentioned above.
+The levels range from `strong positive effect` over `no effect` to `strong negative effect`.
+The `potential` and `unknown` levels end the row.
+
+```{r}
+set.seed(20191126)
+n <- 50
+ds <- data.frame(
+  point_estimate = runif(n, min = -2, max = 2),
+  se = runif(n)
+)
+ds$lcl <- qnorm(p = 0.025, mean = ds$point_estimate, sd = ds$se)
+ds$ucl <- qnorm(p = 0.975, mean = ds$point_estimate, sd = ds$se)
+library(effectclass)
+ds$effect <- classification(lcl = ds$lcl, ucl = ds$ucl, threshold = 1)
+table(ds$effect)
```

Comment:
Maybe show `kable(ds)` instead of `table(ds$effect)`? I think this is easier to follow. In that case consider using the code you used on lines 187-192 to demonstrate the functionality, rather than using `runif()` (in order to restrict the number of rows of the table)

Reply:
Updated in a7e65cc